compass/Coordinator.java

```diff
@@ -77,16 +79,44 @@ public Coordinator(CoordinatorConfiguration config, SignatureSource signatureSou
     }).collect(Collectors.toList());
   }
 
+  private static CoordinatorState loadState() throws IOException, ClassNotFoundException {
+    CoordinatorState state;
+    ObjectInputStream ois = new ObjectInputStream(new FileInputStream(CoordinatorState.COORDINATOR_STATE_PATH));
+    state = (CoordinatorState) ois.readObject();
+    ois.close();
+    return state;
+  }
+
+  private void storeState(CoordinatorState state) throws IOException {
+    ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(CoordinatorState.COORDINATOR_STATE_PATH));
+    oos.writeObject(state);
+    oos.close();
+  }
+
   public static void main(String[] args) throws Exception {
     CoordinatorConfiguration config = new CoordinatorConfiguration();
+    CoordinatorState state;
+    // We want an empty state if bootstrapping
+    if (config.bootstrap) {
+      state = new CoordinatorState();
+    } else {
+      try {
+        state = loadState();
+      } catch (Exception e) {
+        String msg = "Error loading Compass state file! This should not happen unless when bootstrapping.";
+
+        log.error(msg);
```

Comment:
:+1:

Reply:
I would still put the error in the log be cause it can still give extra info about why the file failed to load.
You can do `log.error(msg, e)` or if you don't want the stack trace `log.error(msg + " e.getMessage()")'.

Reply:
This can also be done later (low importance)